Pequenos ajustes no projeto "NFe.Integracao"

AndrewsFB opened the pull request:
- Alguns bugs foram resolvidos.
- Adicionada a possibilidade de emitir NFC-e para MG.
- Foi criado um recurso para imprimir o NFC-e.

Commit:
Atualizando o fork.

Commit:
A configuração do certificado no projeto NFe.Integracao passou a considerar o atributo "TipoCertificado". O projeto não estava funcional por causa deste problema.

Commit:
Melhorias

- Acrescentando URL´s do NFC-e para o estado MG. (Homologação)
- Definindo protocolo TLS como padrão no projeto "NFe.Integração".

Commit:
Melhorias

- Foram acrescentandas as rotas que faltavam na classe "Enderecador" para que o NFC-e funciona-se no estado de MG.

Commit:
Melhorias

- Fixando o uso do certificado A1 para o projeto "NFe.Integração". Isso é feito para que o exe funcione em qualquer máquina de uma rede interna quando o certificado for depositado no servidor.
- Foi acrescentada uma instrução para ler uma tecla ao executar o aplicativo quando em modo de debug. Isso é feito para que a aplicação não feche ao terminar sua execução.

Commit:
Melhorias

- A opção de imprimir um NFC-e a partir do seu xml foi acrescentada ao projeto "NFe.Integração".

marcosgerene commented:
Por favor, resolva os conflitos, basicamente são versões do Assembly

Commit:
Finalizando resolução de conflitos.

AndrewsFB commented:
Feito.

marcosgerene commented on NFe.Integracao/NFeFacade.cs:
```diff
@@ -166,20 +172,22 @@ private void CarregarConfiguracoes()
+            ConfiguracaoServico.Instancia.VersaoNFeRetAutorizacao = VersaoServico.ve400;
+            ConfiguracaoServico.Instancia.VersaoNfeRetRecepcao = VersaoServico.ve400;
+            ConfiguracaoServico.Instancia.VersaoNfeStatusServico = VersaoServico.ve400;
+            ConfiguracaoServico.Instancia.VersaoRecepcaoEventoCceCancelamento = VersaoServico.ve400;
```
As duas versões ainda funcionam, seria interessante criar uma opção para definir qual usar...

marcosgerene commented on NFe.Integracao/Program.cs:
```diff
@@ -548,6 +565,27 @@ private static void EnviarNFe(string dadosDoEnvio)
+        /// <summary>
+        /// Imprime um NFC-e em um arquivo jpeg.
+        /// </summary>
+        /// <param name="dadosDaImpressao">Informações da impressão no formato: pathDoArquivoXml#localOndeSalvarOJpeg</param>
```
Aqui pelo que vi da assinautra do método é "pathDoArquivoXml#localOndeSalvarOJpeg#idToken#Csc"

Commit:
Melhorias/Correções no projeto "NFe.Integração"

- A versão da NFe agora pode ser definida ao usar a opção configurar (Somente 3.10 e 4.00)

- O comentário de documentação do método "ImprimirNFCe" na classe "Program" foi corrigido.

AndrewsFB commented:
Feito.

marcosgerene commented:
Pra mim está ok, não vou aceitar o pull porque não uso o NFe.Integracao, fica dificil eu analisar mais à fundo, vou aguardar o @robertorp  ou o @adeniltonbs aceitar ou dar um ok, mas ao meu ver está tudo ok.

AndrewsFB commented:
Tranquilo Marcos. 
Confesso que estranhei bastante a integração, está bem diferente de quando eu construí. Tem até um conceito que foi mudado que fez muita diferença pro meu uso aqui. Depois vou abrir uma issue pra falarmos sobre o assunto...

robertorp commented:
Resolva o conflito para estar aceitando o pull

AndrewsFB commented:
Quais conflitos?

robertorp commented:
no final desse pull olhe a mensagem 

This branch has conflicts that must be resolved

NFe.Integracao/NFeFacade.cs

Commit:
Resolendo conflitos.

AndrewsFB commented:
Feito.

robertorp commented on NFe.Integracao/NFeFacade.cs:
```diff
@@ -166,19 +171,24 @@ private void CarregarConfiguracoes()
+            var versaoNFe = Properties.Settings.Default.versao_NFe;
+
+            ConfiguracaoServico.Instancia.ProtocoloDeSeguranca = System.Net.SecurityProtocolType.Tls;
+            ConfiguracaoServico.Instancia.VersaoNfceAministracaoCSC = versaoNFe;
```
Vou aprovar o pull mas aqui deixo somente uma observação
Agora você pode optar pelo uso de 

`ConfiguracaoServico.Instancia.VersaoLayout = 4.00 ou 3.00` e o próprio zeus se encarrega de selecionar as versões corretas dos serviços

robertorp commented on NFe.Servicos/Enderecador.cs:
```diff
@@ -709,14 +727,14 @@ private static List<EnderecoServico> CarregarUrlsServicos()
-                endServico.Add(new EnderecoServico(ServicoNFe.RecepcaoEventoCartaCorrecao, VersaoServico.ve400, TipoAmbiente.taHomologacao, emissao, Estado.MT, ModeloDocumento.NFe, "https://homologacao.sefaz.mt.gov.br/nfews/v2/services/RecepcaoEvento4"));
-                endServico.Add(new EnderecoServico(ServicoNFe.NFeAutorizacao, VersaoServico.ve400, TipoAmbiente.taHomologacao, emissao, Estado.MT, ModeloDocumento.NFe, "https://homologacao.sefaz.mt.gov.br/nfews/v2/services/NfeAutorizacao4"));
-                endServico.Add(new EnderecoServico(ServicoNFe.NFeRetAutorizacao, VersaoServico.ve400, TipoAmbiente.taHomologacao, emissao, Estado.MT, ModeloDocumento.NFe, "https://homologacao.sefaz.mt.gov.br/nfews/v2/services/NfeRetAutorizacao4"));
+                endServico.Add(new EnderecoServico(ServicoNFe.NfeInutilizacao, VersaoServico.ve400, TipoAmbiente.taHomologacao, emissao, Estado.MT, ModeloDocumento.NFe, "https://hom.nfe.sefaz.ms.gov.br/ws/NFeInutilizacao4"));
```
url errada a correta e a anterior
Observe que as urls anteriores estão para MT
já as novas urls aqui estão para MS

robertorp commented on NFe.Servicos/Enderecador.cs:
```diff
@@ -709,14 +727,14 @@ private static List<EnderecoServico> CarregarUrlsServicos()
-                endServico.Add(new EnderecoServico(ServicoNFe.NFeAutorizacao, VersaoServico.ve400, TipoAmbiente.taHomologacao, emissao, Estado.MT, ModeloDocumento.NFe, "https://homologacao.sefaz.mt.gov.br/nfews/v2/services/NfeAutorizacao4"));
-                endServico.Add(new EnderecoServico(ServicoNFe.NFeRetAutorizacao, VersaoServico.ve400, TipoAmbiente.taHomologacao, emissao, Estado.MT, ModeloDocumento.NFe, "https://homologacao.sefaz.mt.gov.br/nfews/v2/services/NfeRetAutorizacao4"));
+                endServico.Add(new EnderecoServico(ServicoNFe.NfeInutilizacao, VersaoServico.ve400, TipoAmbiente.taHomologacao, emissao, Estado.MT, ModeloDocumento.NFe, "https://hom.nfe.sefaz.ms.gov.br/ws/NFeInutilizacao4"));
+                endServico.Add(new EnderecoServico(ServicoNFe.NfeConsultaProtocolo, VersaoServico.ve400, TipoAmbiente.taHomologacao, emissao, Estado.MT, ModeloDocumento.NFe, "https://hom.nfe.sefaz.ms.gov.br/ws/NFeConsultaProtocolo4"));
```
url errada a correta e a anterior
Observe que as urls anteriores estão para MT
já as novas urls aqui estão para MS

robertorp commented on NFe.Servicos/Enderecador.cs:
```diff
@@ -709,14 +727,14 @@ private static List<EnderecoServico> CarregarUrlsServicos()
-                endServico.Add(new EnderecoServico(ServicoNFe.NFeRetAutorizacao, VersaoServico.ve400, TipoAmbiente.taHomologacao, emissao, Estado.MT, ModeloDocumento.NFe, "https://homologacao.sefaz.mt.gov.br/nfews/v2/services/NfeRetAutorizacao4"));
+                endServico.Add(new EnderecoServico(ServicoNFe.NfeInutilizacao, VersaoServico.ve400, TipoAmbiente.taHomologacao, emissao, Estado.MT, ModeloDocumento.NFe, "https://hom.nfe.sefaz.ms.gov.br/ws/NFeInutilizacao4"));
+                endServico.Add(new EnderecoServico(ServicoNFe.NfeConsultaProtocolo, VersaoServico.ve400, TipoAmbiente.taHomologacao, emissao, Estado.MT, ModeloDocumento.NFe, "https://hom.nfe.sefaz.ms.gov.br/ws/NFeConsultaProtocolo4"));
+                endServico.Add(new EnderecoServico(ServicoNFe.NfeStatusServico, VersaoServico.ve400, TipoAmbiente.taHomologacao, emissao, Estado.MT, ModeloDocumento.NFe, "https://hom.nfe.sefaz.ms.gov.br/ws/NFeStatusServico4"));
```
url errada a correta e a anterior
Observe que as urls anteriores estão para MT
já as novas urls aqui estão para MS

robertorp commented on NFe.Servicos/Enderecador.cs:
```diff
@@ -709,14 +727,14 @@ private static List<EnderecoServico> CarregarUrlsServicos()
+                endServico.Add(new EnderecoServico(ServicoNFe.NfeInutilizacao, VersaoServico.ve400, TipoAmbiente.taHomologacao, emissao, Estado.MT, ModeloDocumento.NFe, "https://hom.nfe.sefaz.ms.gov.br/ws/NFeInutilizacao4"));
+                endServico.Add(new EnderecoServico(ServicoNFe.NfeConsultaProtocolo, VersaoServico.ve400, TipoAmbiente.taHomologacao, emissao, Estado.MT, ModeloDocumento.NFe, "https://hom.nfe.sefaz.ms.gov.br/ws/NFeConsultaProtocolo4"));
+                endServico.Add(new EnderecoServico(ServicoNFe.NfeStatusServico, VersaoServico.ve400, TipoAmbiente.taHomologacao, emissao, Estado.MT, ModeloDocumento.NFe, "https://hom.nfe.sefaz.ms.gov.br/ws/NFeStatusServico4"));
+                endServico.Add(new EnderecoServico(ServicoNFe.NfeConsultaCadastro, VersaoServico.ve400, TipoAmbiente.taHomologacao, emissao, Estado.MT, ModeloDocumento.NFe, "https://hom.nfe.sefaz.ms.gov.br/ws/CadConsultaCadastro4"));
```
url errada a correta e a anterior
Observe que as urls anteriores estão para MT
já as novas urls aqui estão para MS

robertorp commented on NFe.Servicos/Enderecador.cs:
```diff
@@ -709,14 +727,14 @@ private static List<EnderecoServico> CarregarUrlsServicos()
+                endServico.Add(new EnderecoServico(ServicoNFe.NfeConsultaProtocolo, VersaoServico.ve400, TipoAmbiente.taHomologacao, emissao, Estado.MT, ModeloDocumento.NFe, "https://hom.nfe.sefaz.ms.gov.br/ws/NFeConsultaProtocolo4"));
+                endServico.Add(new EnderecoServico(ServicoNFe.NfeStatusServico, VersaoServico.ve400, TipoAmbiente.taHomologacao, emissao, Estado.MT, ModeloDocumento.NFe, "https://hom.nfe.sefaz.ms.gov.br/ws/NFeStatusServico4"));
+                endServico.Add(new EnderecoServico(ServicoNFe.NfeConsultaCadastro, VersaoServico.ve400, TipoAmbiente.taHomologacao, emissao, Estado.MT, ModeloDocumento.NFe, "https://hom.nfe.sefaz.ms.gov.br/ws/CadConsultaCadastro4"));
+                endServico.Add(new EnderecoServico(ServicoNFe.RecepcaoEventoCancelmento, VersaoServico.ve400, TipoAmbiente.taHomologacao, emissao, Estado.MT, ModeloDocumento.NFe, "https://hom.nfe.sefaz.ms.gov.br/ws/NFeRecepcaoEvento4"));
```
url errada a correta e a anterior
Observe que as urls anteriores estão para MT
já as novas urls aqui estão para MS

robertorp commented on NFe.Servicos/Enderecador.cs:
```diff
@@ -709,14 +727,14 @@ private static List<EnderecoServico> CarregarUrlsServicos()
+                endServico.Add(new EnderecoServico(ServicoNFe.NfeStatusServico, VersaoServico.ve400, TipoAmbiente.taHomologacao, emissao, Estado.MT, ModeloDocumento.NFe, "https://hom.nfe.sefaz.ms.gov.br/ws/NFeStatusServico4"));
+                endServico.Add(new EnderecoServico(ServicoNFe.NfeConsultaCadastro, VersaoServico.ve400, TipoAmbiente.taHomologacao, emissao, Estado.MT, ModeloDocumento.NFe, "https://hom.nfe.sefaz.ms.gov.br/ws/CadConsultaCadastro4"));
+                endServico.Add(new EnderecoServico(ServicoNFe.RecepcaoEventoCancelmento, VersaoServico.ve400, TipoAmbiente.taHomologacao, emissao, Estado.MT, ModeloDocumento.NFe, "https://hom.nfe.sefaz.ms.gov.br/ws/NFeRecepcaoEvento4"));
+                endServico.Add(new EnderecoServico(ServicoNFe.RecepcaoEventoCartaCorrecao, VersaoServico.ve400, TipoAmbiente.taHomologacao, emissao, Estado.MT, ModeloDocumento.NFe, "https://hom.nfe.sefaz.ms.gov.br/ws/NFeRecepcaoEvento4"));
```
url errada a correta e a anterior
Observe que as urls anteriores estão para MT
já as novas urls aqui estão para MS

robertorp commented on NFe.Servicos/Enderecador.cs:
```diff
@@ -709,14 +727,14 @@ private static List<EnderecoServico> CarregarUrlsServicos()
+                endServico.Add(new EnderecoServico(ServicoNFe.NfeConsultaCadastro, VersaoServico.ve400, TipoAmbiente.taHomologacao, emissao, Estado.MT, ModeloDocumento.NFe, "https://hom.nfe.sefaz.ms.gov.br/ws/CadConsultaCadastro4"));
+                endServico.Add(new EnderecoServico(ServicoNFe.RecepcaoEventoCancelmento, VersaoServico.ve400, TipoAmbiente.taHomologacao, emissao, Estado.MT, ModeloDocumento.NFe, "https://hom.nfe.sefaz.ms.gov.br/ws/NFeRecepcaoEvento4"));
+                endServico.Add(new EnderecoServico(ServicoNFe.RecepcaoEventoCartaCorrecao, VersaoServico.ve400, TipoAmbiente.taHomologacao, emissao, Estado.MT, ModeloDocumento.NFe, "https://hom.nfe.sefaz.ms.gov.br/ws/NFeRecepcaoEvento4"));
+                endServico.Add(new EnderecoServico(ServicoNFe.NFeAutorizacao, VersaoServico.ve400, TipoAmbiente.taHomologacao, emissao, Estado.MT, ModeloDocumento.NFe, "https://hom.nfe.sefaz.ms.gov.br/ws/NFeAutorizacao4"));
```
url errada a correta e a anterior
Observe que as urls anteriores estão para MT
já as novas urls aqui estão para MS

robertorp commented on NFe.Servicos/Enderecador.cs:
```diff
@@ -709,14 +727,14 @@ private static List<EnderecoServico> CarregarUrlsServicos()
+                endServico.Add(new EnderecoServico(ServicoNFe.RecepcaoEventoCancelmento, VersaoServico.ve400, TipoAmbiente.taHomologacao, emissao, Estado.MT, ModeloDocumento.NFe, "https://hom.nfe.sefaz.ms.gov.br/ws/NFeRecepcaoEvento4"));
+                endServico.Add(new EnderecoServico(ServicoNFe.RecepcaoEventoCartaCorrecao, VersaoServico.ve400, TipoAmbiente.taHomologacao, emissao, Estado.MT, ModeloDocumento.NFe, "https://hom.nfe.sefaz.ms.gov.br/ws/NFeRecepcaoEvento4"));
+                endServico.Add(new EnderecoServico(ServicoNFe.NFeAutorizacao, VersaoServico.ve400, TipoAmbiente.taHomologacao, emissao, Estado.MT, ModeloDocumento.NFe, "https://hom.nfe.sefaz.ms.gov.br/ws/NFeAutorizacao4"));
+                endServico.Add(new EnderecoServico(ServicoNFe.NFeRetAutorizacao, VersaoServico.ve400, TipoAmbiente.taHomologacao, emissao, Estado.MT, ModeloDocumento.NFe, "https://hom.nfe.sefaz.ms.gov.br/ws/NFeRetAutorizacao4"));
```
url errada a correta e a anterior
Observe que as urls anteriores estão para MT
já as novas urls aqui estão para MS

robertorp commented:
Somente mesmo a questão das urls de MT que precisam ser corrigidas na minha visão, para ficar zerado

Commit:
Corrigindo urls do MT que estavam incorretas.

AndrewsFB commented:
Corrigido.

robertorp commented:
obrigado @AndrewsFB  !
